plugins/modules/omevv_baseline_profile.py

```diff
@@ -0,0 +1,589 @@
+        vcenter_uuid = self.module.params.get('vcenter_uuid')
+        response, err_msg = self.omevv_baseline_obj.create_baseline_profile(
+            name=payload.get('name'),
+            firmware_repo_id=payload.get('firmwareRepoId'),
+            group_ids=payload.get('groupIds'),
+            vcenter_uuid=vcenter_uuid,
+            payload=payload
+        )
+        if response.success:
+            profile_resp = self.omevv_baseline_obj.get_baseline_profile_by_id(response.json_data, vcenter_uuid)
+            while profile_resp["status"] not in ["SUCCESSFUL", "FAILED"]:
+                time.sleep(3)
+                profile_resp = self.omevv_baseline_obj.get_baseline_profile_by_id(response.json_data, vcenter_uuid)
+
+            diff = self.diff_mode_check(payload)
+            if self.module._diff and profile_resp["status"] == "SUCCESSFUL":
+                self.module.exit_json(msg=SUCCESS_CREATION_MSG, baseline_profile_info=profile_resp, diff=diff, changed=True)
+            elif profile_resp["status"] == "SUCCESSFUL":
+                self.module.exit_json(msg=SUCCESS_CREATION_MSG, baseline_profile_info=profile_resp, changed=True)
+            else:
+                self.module.exit_json(msg=FAILED_CREATION_MSG, baseline_profile_info=profile_resp, failed=True)
+        else:
+            self.module.exit_json(msg=FAILED_CREATION_MSG, failed=True)
+
+    def execute(self):
+        self.validate_common_params()
+        firmware_repo_id = self.omevv_baseline_obj.get_repo_id_by_name(self.module.params.get('repository_profile'))
+        group_ids = self.omevv_baseline_obj.get_group_ids_for_clusters(
+            vcenter_uuid=self.module.params.get('vcenter_uuid'),
+            cluster_names=self.module.params.get('cluster')
+        )
+        job_schedule = self.omevv_baseline_obj.create_job_schedule(self.module.params.get('days'), self.module.params.get('time'))
+        payload = {
+            "name": self.module.params.get('name'),
+            "firmwareRepoId": firmware_repo_id,
+            "groupIds": group_ids
+        }
+        payload["repository_profile"] = self.module.params.get('repository_profile')
+        description = self.module.params.get('description')
+        if description is not None:
+            payload["description"] = description
+        if job_schedule:
+            payload["jobSchedule"] = job_schedule
+
+        if self.module.check_mode and self.module._diff:
+            diff = self.diff_mode_check(payload)
+            self.module.exit_json(msg=CHANGES_FOUND_MSG, diff=diff, changed=True)
+
+        if self.module.check_mode:
+            self.module.exit_json(msg=CHANGES_FOUND_MSG, changed=True)
+
+        self.perform_create_baseline_profile(payload)
+
+
+class ModifyBaselineProfile(BaselineProfile):
+
+    def __init__(self, module, rest_obj, existing_profile):
+        super().__init__(module, rest_obj)
+        self.existing_profile = existing_profile
+
+    def diff_mode_check(self, payload):
```

Comment:
Again, why we are handling this ourself, why not use recursive_diff?

Reply:
tested it implemented my own logic